### DISCUSSION TITLE
有没有什么方法能够获取到fakedns某个ip对应的域名

### DISCUSSION THREAD

**Discussion by ggggle** [XTLS/Xray-core#490] [Q&A]
有些时候某些域名通过fakedns解析了，但是路由的时候走的是直连，导致连接失败。
已知可以把日志级别调到info，但是日志太多因此一般都开的warning。所以有没有api接口支持查询呢？

**Comment by rurirei** [comment]
> 有些时候某些域名通过fakedns解析了，但是路由的时候走的是直连，导致连接失败。

?

**Reply by ggggle** [reply]
比如给www.google.com分配了198.18.146.207，想要通过198.18.146.207反查到对应的domain是www.google.com

**Reply by rurirei** [reply]
how you met the connection error?

**Reply by ggggle** [reply]
一般是国内dns解析超时，轮询到fakedns分配了ip，最终路由的时候从direct路由出去了
